api/groups/blockGroup.go

```diff
@@ -0,0 +1,174 @@
+	}
+
+	bg := &blockGroup{
+		facade:    facade,
+		baseGroup: &baseGroup{},
+	}
+
+	endpoints := []*shared.EndpointHandlerData{
+		{
+			Path:    getBlockByNoncePath,
+			Method:  http.MethodGet,
+			Handler: bg.getBlockByNonce,
+		},
+		{
+			Path:    getBlockByHashPath,
+			Method:  http.MethodGet,
+			Handler: bg.getBlockByHash,
+		},
+	}
+	bg.endpoints = endpoints
+
+	return bg, nil
+}
+
+func (bg *blockGroup) getBlockByNonce(c *gin.Context) {
+	nonce, err := getQueryParamNonce(c)
+	if err != nil {
+		shared.RespondWithValidationError(
+			c, fmt.Sprintf("%s: %s", errors.ErrValidation.Error(), errors.ErrInvalidBlockNonce.Error()),
+		)
+		return
+	}
+
+	withTxs, err := getQueryParamWithTxs(c)
+	if err != nil {
+		shared.RespondWithValidationError(
+			c, fmt.Sprintf("%s: %s", errors.ErrValidation.Error(), errors.ErrInvalidQueryParameter.Error()),
+		)
+		return
+	}
+
+	start := time.Now()
+	block, err := bg.getFacade().GetBlockByNonce(nonce, withTxs)
+	log.Debug(fmt.Sprintf("GetBlockByNonce took %s", time.Since(start)))
+	if err != nil {
+		shared.RespondWith(
+			c,
+			http.StatusInternalServerError,
+			nil,
+			fmt.Sprintf("%s: %s", errors.ErrGetBlock.Error(), err.Error()),
+			shared.ReturnCodeInternalError,
+		)
+		return
+	}
+
+	shared.RespondWith(c, http.StatusOK, gin.H{"block": block}, "", shared.ReturnCodeSuccess)
+
+}
+
+func (bg *blockGroup) getBlockByHash(c *gin.Context) {
+	hash := c.Param("hash")
+	if hash == "" {
+		shared.RespondWithValidationError(
+			c, fmt.Sprintf("%s: %s", errors.ErrValidation.Error(), errors.ErrValidationEmptyBlockHash.Error()),
+		)
+		return
+	}
+
+	withTxs, err := getQueryParamWithTxs(c)
+	if err != nil {
+		shared.RespondWithValidationError(
+			c, fmt.Sprintf("%s: %s", errors.ErrValidation.Error(), errors.ErrInvalidBlockNonce.Error()),
+		)
+		return
+	}
+
+	start := time.Now()
+	block, err := bg.getFacade().GetBlockByHash(hash, withTxs)
+	log.Debug(fmt.Sprintf("GetBlockByHash took %s", time.Since(start)))
+	if err != nil {
+		shared.RespondWith(
+			c,
+			http.StatusInternalServerError,
+			nil,
+			fmt.Sprintf("%s: %s", errors.ErrGetBlock.Error(), err.Error()),
+			shared.ReturnCodeInternalError,
+		)
+		return
+	}
+
+	shared.RespondWith(c, http.StatusOK, gin.H{"block": block}, "", shared.ReturnCodeSuccess)
+}
+
+func getQueryParamWithTxs(c *gin.Context) (bool, error) {
+	withTxsStr := c.Request.URL.Query().Get("withTxs")
+	if withTxsStr == "" {
+		return false, nil
+	}
+
+	return strconv.ParseBool(withTxsStr)
+}
+
+func getQueryParamNonce(c *gin.Context) (uint64, error) {
+	nonceStr := c.Param("nonce")
+	if nonceStr == "" {
+		return 0, errors.ErrInvalidBlockNonce
+	}
+
+	return strconv.ParseUint(nonceStr, 10, 64)
+}
+
+func (bg *blockGroup) getFacade() blockFacadeHandler {
+	bg.mutFacade.RLock()
+	defer bg.mutFacade.RUnlock()
+
+	return bg.facade
+}
+
+// UpdateFacade will update the facade
+func (bg *blockGroup) UpdateFacade(newFacade interface{}) error {
+	if newFacade == nil {
```

Comment:
unit tests on all untested UpdateFacade functions.

Reply:
I would leave this for the next PR(s) as there are also other components that need to be unit tested

Reply:
great